app/main.py

```diff
@@ -1,6 +1,17 @@
-from typing import Callable
+from typing import Callable, Any
+from functools import wraps
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    database = {}
+
+    @wraps(func)
+    def wrapper(*args, **kwargs) -> Any:
+        if f"{args}, {kwargs}" in database.keys():
+            print("Getting from cache")
+            return database[f"{args}, {kwargs}"]
+        else:
+            print("Calculating new result")
+            database[f"{args}, {kwargs}"] = func(*args, **kwargs)
+            return database[f"{args}, {kwargs}"]
```

Comment:
return the same in both cases
Move it after if-else statement

Reply:
ok